eth_rlp/main.py

```diff
@@ -93,3 +93,9 @@ def __iter__(self):
             return iter(getattr(self, field) for field, _ in self.fields)
         else:
             return super().__iter__()
+
+    def as_dict(self):
+        if hasattr(super(), 'as_dict'):
```

Comment:
maybe go with an *optimistic* approach since people are more and more likely to have the `1.x` version installed.

```
try:
    return self.as_dict()
except AttributeError:
    return vars(self)
```

Performance hit only happens in `rlp<1.0`